### PR TITLE
Binary format type for ply IO

### PR DESCRIPTION
**Pull request by Kokika**
## Description

This PR writes binary PLY files instead of ASCII ones if `.bin.` is detected in the name of the file to write.

### CODE REVIEW

File: src/aliceVision/sfmDataIO/plyIO.cpp [hunk 1]
```diff
@@ -53,36 +60,61 @@ bool savePLY(
...
-              << " 0 255 0" << "\n";
-          }
+            for (const auto& view : sfmData.getViews())
+            {
+                if (sfmData.isPoseAndIntrinsicDefined(view.second.get()))
+                {
+                    const geometry::Pose3 pose = sfmData.getPose(*(view.second.get())).getTransform();
+                    if (b_binary)
+                    {
+                        auto point = pose.center().transpose();
```
Comment (servantftransperfect) [inline src/aliceVision/sfmDataIO/plyIO.cpp]: Why transposing the vector ?

File: src/aliceVision/sfmDataIO/plyIO.cpp [hunk 2]
```diff
@@ -53,36 +60,61 @@ bool savePLY(
...
-                 << (int)iterLandmarks->second.rgb.b() << "\n";
+        if (b_structure)
+        {
+            const sfmData::Landmarks& landmarks = sfmData.getLandmarks();
+            for(sfmData::Landmarks::const_iterator iterLandmarks = landmarks.begin(); iterLandmarks != landmarks.end();
+                ++iterLandmarks)
+            {
+                if (b_binary)
+                {
+                    const auto& point = iterLandmarks->second.X.transpose();
```
Comment (servantftransperfect) [inline src/aliceVision/sfmDataIO/plyIO.cpp]: Idem, why transposing ?

File: src/aliceVision/sfmDataIO/plyIO.cpp [hunk 3]
```diff
@@ -17,34 +17,41 @@ bool savePLY(
...
+        IndexT view_with_pose_count = 0;
+        if (b_extrinsics)
+        {
+            for (const auto& view : sfmData.getViews())
+            {
+                view_with_pose_count += sfmData.isPoseAndIntrinsicDefined(view.second.get());
+            }
+        }
+        stream << "ply"
+            << '\n' << "format " << (b_binary ? "binary_little_endian" : "ascii") << " 1.0 "
```
Comment (servantftransperfect) [inline src/aliceVision/sfmDataIO/plyIO.cpp]: ok, i'm confident we're not going to use big endian anytime soon :)

File: src/aliceVision/sfmDataIO/plyIO.cpp [hunk 4]
```diff
@@ -53,36 +60,61 @@ bool savePLY(
       << '\n' << "property uchar blue"
       << '\n' << "end_header" << std::endl;
 
-      if (b_extrinsics)
-      {
-        for (const auto& view : sfmData.getViews())
+        if (b_extrinsics)
```
Comment (servantftransperfect) [inline src/aliceVision/sfmDataIO/plyIO.cpp]: I know this is not part of your PR. However, do you need the camera position to be stored ?

**Review by servantftransperfect — COMMENTED** [review]
Working. Waiting for answer.

File: src/aliceVision/sfmDataIO/plyIO.cpp (hunk 4 above)
Comment (Kokika) [reply src/aliceVision/sfmDataIO/plyIO.cpp]: I don't need it.

Comment (Kokika) [reply src/aliceVision/sfmDataIO/plyIO.cpp]: De hozzátettem a kompatibilitás miatt

**Review by cbentejac — DISMISSED** [review]

**Comment by cbentejac** [comment]
Hi @Kokika,

Thank you for your contribution! 

I rebased your branch on top of `develop` and fixed the merge conflicts. We'll merge it as soon as the CI builds are done!

**Review by cbentejac — APPROVED** [review]

**Pull request merged**